lms/urls.py

```diff
@@ -613,6 +613,17 @@
         url(r'^certificates/html', 'certificates.views.render_html_view', name='cert_html_view'),
     )
 
+# Access to courseware as an LTI provider
+if settings.FEATURES["ENABLE_LTI_PROVIDER"]:
+    urlpatterns += (
+        url(r'^courses/{}/lti_launch/(?P<chapter>[^/]*)/(?P<section>[^/]*)/(?P<position>[^/]*)/?$'.format(settings.COURSE_ID_PATTERN),
+            'lti_provider.views.lti_launch', name="lti_provider_launch"),
+    )
+    urlpatterns += (
+        url(r'^lti_provider/lti_run/',
+            'lti_provider.views.lti_run', name="lti_provider_run"),
+    )
```

Comment:
This also enables you to arbitrarily add new endpoints without growing the base file.